Sources/StreamChatUI/CommonViews/AvatarView/ChatChannelAvatarView.swift

```diff
@@ -30,31 +47,293 @@ open class ChatChannelAvatarView: _View, ThemeProvider, SwiftUIRepresentable {
+    /// - Parameter channel: The channel
+    open func loadDirectMessageChannelAvatar(channel: ChatChannel) {
+        let lastActiveMembers = channel.lastActiveMembers
+            .sorted { $0.memberCreatedAt < $1.memberCreatedAt }
+            .filter { $0.id != content.currentUserId }
+        
+        // If there are no members other than the current user in the channel, load a placeholder
+        if lastActiveMembers.isEmpty {
+            presenceAvatarView.avatarView.imageView.loadImage(
+                from: nil,
+                placeholder: appearance.images.userAvatarPlaceholder4,
+                preferredSize: .avatarThumbnailSize,
+                components: components
+            )
+            return
+        }
+        
+        let otherMember = lastActiveMembers.first
+        let isOnlineIndicatorVisible = otherMember?.isOnline ?? false
+        
+        presenceAvatarView.avatarView.imageView.loadImage(
+            from: otherMember?.imageURL,
+            placeholder: appearance.images.userAvatarPlaceholder3,
+            preferredSize: .avatarThumbnailSize,
+            components: components
+        )
+        
         presenceAvatarView.isOnlineIndicatorVisible = isOnlineIndicatorVisible
     }
+    
+    /// Loads an avatar which is merged (tiled) version of the first four active members of the channel
+    /// - Parameter channel: The channel
+    open func loadMergedAvatars(channel: ChatChannel) {
+        // The channel is a non-DM channel, hide the online indicator
+        presenceAvatarView.isOnlineIndicatorVisible = false
+        
+        let lastActiveMembers = channel.lastActiveMembers
+            .sorted { $0.memberCreatedAt < $1.memberCreatedAt }
+            .filter { $0.id != content.currentUserId }
+        
+        // If there are no members other than the current user in the channel, load a placeholder
+        if lastActiveMembers.isEmpty {
+            presenceAvatarView.avatarView.imageView.loadImage(
+                from: nil,
+                placeholder: appearance.images.userAvatarPlaceholder4,
+                preferredSize: .avatarThumbnailSize,
+                components: components
+            )
+            return
+        }
+        
+        var urls = lastActiveMembers.map(\.imageURL)
+        
+        if urls.isEmpty {
+            presenceAvatarView.avatarView.imageView.loadImage(
+                from: nil,
+                placeholder: appearance.images.userAvatarPlaceholder3,
+                preferredSize: .avatarThumbnailSize,
+                components: components
+            )
+            return
+        }
+        
+        // We show a combination of at max 4 images combined
+        urls = Array(urls.prefix(4))
+        
+        guard !SystemEnvironment.isTests else {
+            // When running tests, we load the images synchronously
+            let images = urls.map { UIImage(data: try! Data(contentsOf: $0!))! }
```

Comment:
force unwraps should be removed please use `try?` and guard-if-let chain instead of force unwrapping

Reply:
This code was inside Tests code, i.e. `SystemEnvironment.isTests` hence the force unwrapping. Let's decide on the placement of the test code then.. because Snapshot tests need the images to be loaded synchronously in order to capture a snapshot